OKlib.js

```diff
@@ -824,416 +767,378 @@ registerPlugin({
-     * @param  {Object[] | Object} element
-     * @return {Object[]} An Array containing the given Elements
+     * creates an array containing the given element or returns immediately if an array is given
+     * @param {Object[] | Object} element
+     * @return {Object[]} an array containing the given elements
      */
     function arrayCreateArray(element) {
         if (!Array.isArray(element)) {
             let array = [];
-            if (element) {
-                array.push(element);
-            }
+            if (element) array.push(element);
             return array;
         } else {
             return element;
         }
     }
 
     /**
-     * Creates a Set of the given Array, removing any Duplicates
-     * @param  {Object[]} array
-     * @param  {Function} [compare=equal] - Compare function that should be used for the comparison. Default is 'equal'
-     * @return {Object[]}  The Set representation of the given Array
+     * creates a set of the given array, removing any duplicates
+     * @param {Object[]} array
+     * @param {Function} compare a compare function that should be used for the comparison | if not set, 'equal' is used
+     * @return {Object[]} the set representation of the given array
      */
     function arrayCreateSet(array, compare) {
         let result = [];
         if (!Array.isArray(array)) {
             result.push(array);
             return result;
         } else {
-            for (var i = 0; i < array.length; i++) {
-                if (!arrayContainsElement(result, array[i], compare)) {
-                    result.push(array[i]);
-                }
+            for (let i in array) {
+                if (!arrayContainsElement(result, array[i], compare)) result.push(array[i]);
             }
             return result;
         }
     }
 
     /**
-     * Returns the Elements that differ are contained in only one of the given Arrays
-     * @param  {Object[]} array
-     * @param  {Object[]} elements
-     * @param  {Function} compare A Compare Function that should be used for the Comparison, if not set 'equal' is used
-     * @return {Object[]}    An Array that contains the Elements that were only present in one of the Arrays
+     * returns the elements that differ are contained in only one of the given arrays
+     * @param {Object[]} array
+     * @param {Object[]} elements
+     * @param {Function} compare a compare function that should be used for the comparison | if not set, 'equal' is used
+     * @return {Object[]} an array that contains the elements that were only present in one of the arrays
      */
     function arrayDifference(array, elements, compare) {
         let result = [];
-        for (let i = 0; i < elements.length; i++) {
-            if (!arrayContainsElement(array, elements[i], compare)) {
-                result.push(elements[i]);
-            }
+        for (let i in elements) {
```

Comment:
change this to for..of or revert to normal for loop, because for..in is not safe for array operation (might not iterate in the right order)